Applications/Spire/Include/Spire/OrderImbalanceIndicator/CurrentOrderImbalanceIndicatorTableModel.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef SPIRE_CURRENT_ORDER_IMBALANCE_INDICATOR_TABLE_MODEL_HPP
+#define SPIRE_CURRENT_ORDER_IMBALANCE_INDICATOR_TABLE_MODEL_HPP
+#include <Beam/Queues/CallbackQueue.hpp>
+#include <Beam/Threading/TimerBox.hpp>
+#include <Beam/TimeService/TimeClientBox.hpp>
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorTableModel.hpp"
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorModel.hpp"
+
+namespace Spire {
+
+  /**
+   * Implements a TableModel that contains OrderImbalances published within
+   * a dynamic interval between a duration from the present and the present
+   * time.
+   */
+  class CurrentOrderImbalanceIndicatorTableModel : public TableModel {
+    public:
+
+      /**
+       * Represents a function for creating TimeBox instances.
+       * @param duration The duration between starting the TimerBox instance
+       *                 and its expiration.
+       * @returns The TimerBox instance.
+       */
+      using TimerFactory = std::function<
+        Beam::Threading::TimerBox (boost::posix_time::time_duration duration)>;
+
+      /**
+       * Constructs a CurrentOrderImbalanceIndicatorTableModel.
+       * @param offset The offset from the present time.
+       * @param clock The clock used to determine the present time.
+       * @param timer_factory Function used to create the timers used
+       *                      to remove expired OrderImbalances.
+       * @param source The source OrderImbalance model.
+       */
+      CurrentOrderImbalanceIndicatorTableModel(
+        boost::posix_time::time_duration offset,
+        Beam::TimeService::TimeClientBox clock, TimerFactory timer_factory,
+        std::shared_ptr<OrderImbalanceIndicatorModel> source);
+
+      int get_row_size() const override;
+
+      int get_column_size() const override;
+
+      const std::any& at(int row, int column) const override;
+
+      boost::signals2::connection connect_operation_signal(
+        const OperationSignal::slot_type& slot) const override;
+
+    private:
+      struct Expiring {
```

Comment:
Types are nouns, not verbs. It's not even clear what could be meant by an object called ```Expiring``` since that's a state.